### ISSUE TITLE
[bug]: ts-sdk records keploy-testcase even on test mode

### ISSUE THREAD

**Issue by re-Tick** [keploy/typescript-sdk#42]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

 When, I run the application on test mode, and do an API call on endpoint then this will record testcases. 

 Ideally it should just return the responses and not record Keploy-testcases on test mode.

### Steps to reproduce

1. Run the self-hosted Keploy server. Follow the [Quick installation](https://github.com/keploy/keploy#quick-installation) guide
2. Integrate ts-sdk or use the [sample-typescript ](https://github.com/keploy/samples-typescript/tree/main/node-fetch)application.
3. Run the application on test mode.
4. Do an API call on the endpoint of running application.
5. A Keploy-testcase is generated

### Environment

None

### Version

Cloud

**Comment by Frey0-0** [comment]
Hello @re-Tick please assign me this issue! I can't assign it to myself for some reason.

**Comment by re-Tick** [comment]
Done. Looking forward for your PR

**Comment by shreyanshshah27** [comment]
Hey @re-Tick 

I've tried to fix this issue. It would be great if you can review the PR.

**Comment by shreyanshshah27** [comment]
Hey @Frey0-0 

My sincere apologies, I had already stated working on this issue and by the time I notice that, this issue has been assigned to you, I had already made the fix. I hope you can understand.

**Comment by re-Tick** [comment]
Hi @shreyanshshah27, please look for unassigned issues. Hey @Frey0-0, don't worry I will be adding some more GFI's for TS-SDK, you can pick from them.

This changes looks good to me, have tried to make API call during test mode.

**Comment by shreyanshshah27** [comment]
@re-Tick Definitely, I'll take care of it from the next time.

Yes, I've tried making API call during test mode and those weren't getting recorded.

**Comment by Frey0-0** [comment]
Yeah sure @re-Tick, not a problem I have my uni exams(last night preps you know how it is) coming up so I will get back to work next weekend :))
